From d95eddd49ab5988ba6b2cb98d262b0d573929139 Mon Sep 17 00:00:00 2001
From: Marta Kuczora <kuczoram@cloudera.com>
Date: Mon, 19 Mar 2018 12:03:00 +0100
Subject: [PATCH 1344/1363] Revert "CDH-57844 HIVE-6727 : Table level stats
 for external tables are set incorrectly (Bing Li
 via Ashutosh Chauhan)"

This reverts commit 7be41c7acd5a1e9bdb34241c90e7923f2c532fd6.

Change-Id: Ieb693c3957ae637c4f420dfc50ba89e9bdbb7476
(cherry picked from commit 79f71cc0eae8441341bba79855f2ce13ebd781c8)
---
 .../apache/hadoop/hive/metastore/Warehouse.java    |    2 +-
 .../encryption_join_unencrypted_tbl.q.out          |    8 ++++----
 .../clientpositive/spark/stats_noscan_2.q.out      |    8 ++++----
 .../results/clientpositive/stats_noscan_2.q.out    |    8 ++++----
 4 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/Warehouse.java b/metastore/src/java/org/apache/hadoop/hive/metastore/Warehouse.java
index 79370e4..fcf86c0 100755
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/Warehouse.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/Warehouse.java
@@ -545,7 +545,7 @@ public static String makePartName(List<FieldSchema> partCols,
    */
   public FileStatus[] getFileStatusesForUnpartitionedTable(Database db, Table table)
       throws MetaException {
-    Path tablePath = getDnsPath(new Path(table.getSd().getLocation()));
+    Path tablePath = getTablePath(db, table.getTableName());
     try {
       FileSystem fileSys = tablePath.getFileSystem(conf);
       return HiveStatsUtils.getFileStatusRecurse(tablePath, -1, fileSys);
diff --git a/ql/src/test/results/clientpositive/encrypted/encryption_join_unencrypted_tbl.q.out b/ql/src/test/results/clientpositive/encrypted/encryption_join_unencrypted_tbl.q.out
index 6480e0f..706a516 100644
--- a/ql/src/test/results/clientpositive/encrypted/encryption_join_unencrypted_tbl.q.out
+++ b/ql/src/test/results/clientpositive/encrypted/encryption_join_unencrypted_tbl.q.out
@@ -624,13 +624,13 @@ STAGE PLANS:
               columns.types int:string
 #### A masked pattern was here ####
               name default.encrypted_table
-              numFiles 1
+              numFiles 0
               numRows 0
               rawDataSize 0
               serialization.ddl struct encrypted_table { i32 key, string value}
               serialization.format 1
               serialization.lib org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
-              totalSize 5812
+              totalSize 0
 #### A masked pattern was here ####
             serde: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
           
@@ -644,13 +644,13 @@ STAGE PLANS:
                 columns.types int:string
 #### A masked pattern was here ####
                 name default.encrypted_table
-                numFiles 1
+                numFiles 0
                 numRows 0
                 rawDataSize 0
                 serialization.ddl struct encrypted_table { i32 key, string value}
                 serialization.format 1
                 serialization.lib org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
-                totalSize 5812
+                totalSize 0
 #### A masked pattern was here ####
               serde: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
               name: default.encrypted_table
diff --git a/ql/src/test/results/clientpositive/spark/stats_noscan_2.q.out b/ql/src/test/results/clientpositive/spark/stats_noscan_2.q.out
index b928180..affd43e 100644
--- a/ql/src/test/results/clientpositive/spark/stats_noscan_2.q.out
+++ b/ql/src/test/results/clientpositive/spark/stats_noscan_2.q.out
@@ -50,10 +50,10 @@ Table Type:         	EXTERNAL_TABLE
 Table Parameters:	 	 
 	COLUMN_STATS_ACCURATE	true                
 	EXTERNAL            	TRUE                
-	numFiles            	1                   
+	numFiles            	0                   
 	numRows             	-1                  
 	rawDataSize         	-1                  
-	totalSize           	11                  
+	totalSize           	0                   
 #### A masked pattern was here ####
 	 	 
 # Storage Information	 	 
@@ -94,10 +94,10 @@ Table Type:         	EXTERNAL_TABLE
 Table Parameters:	 	 
 	COLUMN_STATS_ACCURATE	true                
 	EXTERNAL            	TRUE                
-	numFiles            	1                   
+	numFiles            	0                   
 	numRows             	6                   
 	rawDataSize         	6                   
-	totalSize           	11                  
+	totalSize           	0                   
 #### A masked pattern was here ####
 	 	 
 # Storage Information	 	 
diff --git a/ql/src/test/results/clientpositive/stats_noscan_2.q.out b/ql/src/test/results/clientpositive/stats_noscan_2.q.out
index b928180..affd43e 100644
--- a/ql/src/test/results/clientpositive/stats_noscan_2.q.out
+++ b/ql/src/test/results/clientpositive/stats_noscan_2.q.out
@@ -50,10 +50,10 @@ Table Type:         	EXTERNAL_TABLE
 Table Parameters:	 	 
 	COLUMN_STATS_ACCURATE	true                
 	EXTERNAL            	TRUE                
-	numFiles            	1                   
+	numFiles            	0                   
 	numRows             	-1                  
 	rawDataSize         	-1                  
-	totalSize           	11                  
+	totalSize           	0                   
 #### A masked pattern was here ####
 	 	 
 # Storage Information	 	 
@@ -94,10 +94,10 @@ Table Type:         	EXTERNAL_TABLE
 Table Parameters:	 	 
 	COLUMN_STATS_ACCURATE	true                
 	EXTERNAL            	TRUE                
-	numFiles            	1                   
+	numFiles            	0                   
 	numRows             	6                   
 	rawDataSize         	6                   
-	totalSize           	11                  
+	totalSize           	0                   
 #### A masked pattern was here ####
 	 	 
 # Storage Information	 	 
-- 
1.7.9.5

